### PR TITLE
Osmnx merge & new notebook & updated readme

### PR DESCRIPTION
**Pull request by MrJake222**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,17 +31,28 @@ which takes several parameters:
 - `no_of_generations: int` - number of generations to simulate,
 - `report_every_n: int` - report every n-th generation,
 - `report_show: bool` - if `True` then the map image is displayed, otherwise it is saved to the file
+- `simulation_params: SimulationParams` - optional simulation params
 #### Running the app
-To run the app, run the following command in `src` directory:
+To run the app, run the following command in repo's root directory:
+```
+PYTHONPATH="src" python src/main.py
```
Comment (Wisien999) [inline README.md:38]: `PYTHONPATH="${PYTHONPATH}:$(pwd)/src"` is safer

File: README.md [hunk 2]
```diff
@@ -28,17 +31,28 @@ which takes several parameters:
...
+#### Results
+The graphs will be saved to `results` folder in the SVG format.
 
 ### Grid search
 #### Running the grid search
-To run grid search, run the following command in `src` directory:
+To run grid search, run the following command in repo's root directory:
 ```
-python src/grid_search.py
+PYTHONPATH="src" python src/grid_search.py
```
Comment (Wisien999) [inline README.md:55]: `PYTHONPATH="${PYTHONPATH}:$(pwd)/src"` is safer

File: src/grid_search.py [hunk 3]
```diff
@@ -160,100 +160,98 @@ def process_params(tasks, results, G, best_paths, INITIAL_POPULATIONS):
...
+    if cpu_count is not None:
+        parallel_units = cpu_count
+
+    print("Parallel units:", parallel_units)
+
+    # fixed order list
+    # order of keys in params_keys define column order in csv file
+    # must stay the same for all processes
+    params_keys = list(grid_search_params.keys())
+    queue: "Queue[dict]" = Queue()
```
Comment (Wisien999) [inline src/grid_search.py:209]: It somehow works for me, but WTF is That type? Why not Just `Queue[dict[str, Any]]`?

File: src/grid_search.py [hunk 4]
```diff
@@ -160,100 +160,98 @@ def process_params(tasks, results, G, best_paths, INITIAL_POPULATIONS):
...
+    results: "Queue[Tuple[dict, float]]" = Queue()
```
Comment (Wisien999) [inline src/grid_search.py:210]: same and `Tuple` is old, use `tuple`

File: src/grid_search.py (hunk 3 above)
Comment (MrJake222) [reply src/grid_search.py:209]: @kosmydel

File: src/grid_search.py (hunk 4 above)
Comment (MrJake222) [reply src/grid_search.py:210]: @kosmydel

**Pull request merged**